src/js/bg/plugins/calls/call/sip.js

```diff
@@ -180,27 +190,34 @@ class CallSIP extends Call {
             this.session.bye()
         })
 
-        this.session.on('failed', (message) => {
-            this.app.logger.info(`${this}call declined: ${message.status_code}/${this.state.stats.callId}`)
+        this.session.on('failed', (response) => {
             this.busyTone.play()
 
-            if (message.status_code === 480) {
-                // Temporarily Unavailable; Callee currently unavailable.
-                this.setState({status: 'callee_unavailable'})
-            } else if (message.status_code === 486) {
-                // Busy here; Callee is busy.
-                this.setState({status: 'callee_busy'})
-            } else if (message.status_code === 487) {
-                // Request terminated; Request has terminated by bye or cancel.
-                this.setState({status: 'request_terminated'})
+            // See https://sipjs.com/api/0.11.0/session/#failed
+            // The response parameter should be a `SIP.IncomingResponse`. 
+            // The type is not consistent though. It can also be a SIP header 
+            // string or just null.     
+            if (response) {
+                this.app.logger.info(`${this}call declined: ${response.status_code}/${this.state.stats.callId}`)
+                if (response.status_code === 480) {
+                    // Temporarily Unavailable; Callee currently unavailable.
+                    this.setState({status: 'callee_unavailable'})
+                } else if (response.status_code === 486) {
+                    // Busy here; Callee is busy.
+                    this.setState({status: 'callee_busy'})
+                } else if (response.status_code === 487) {
+                    // Request terminated; Request has terminated by bye or cancel.
+                    this.setState({status: 'request_terminated'})
+                } else {
+                    // Assume `request_terminated`, but log unhandled status as a warning.
+                    this.app.logger.warn(`${this}unhandled status code: ${response.status_code}`)
+                    this.setState({status: 'request_terminated'})
+                }
             } else {
-                // Assume `request_terminated`, but log unhandled status as a warning.
-                this.app.logger.warn(`${this}unhandled status code: ${message.status_code}`)
```

Comment:
Maybe good to log the reason of response being `null`, according to the documentation:

> The received response. On a failure not due to a SIP message, this will be null.